### PR TITLE
core: Add a couple of small FF-A messages support for SPs

### PR DESCRIPTION
**Pull request by jellesels-arm**
Add the following functionality:
FFA_ID_GET: Allow a SP to retrieve it's own id
FFA_VERSION: Allow a SP to retrieve the FF-A version supported
FFA_FEATURES: Allow a SP to retrieve the FF-A features supported.

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` 
for commit "core: Add FFA_VERSION handling for SPs"
and commit "core: Add FFA_FEATURES handling for SPs".

File: core/arch/arm/kernel/spmc_sp_handler.c [hunk 1]
```diff
@@ -234,6 +234,11 @@ void spmc_sp_msg_handler(struct thread_smc_args *args,
 			ts_pop_current_session();
 			sp_enter(args, caller_sp);
 			break;
+		case FFA_ID_GET:
+			args->a0 = caller_sp->endpoint_id;
```
[the remark this answers is not in the document]
Comment (jellesels-arm) [reply core/arch/arm/kernel/spmc_sp_handler.c:267]: You are right. Changed it.

**Comment by jforissier** [comment]
Only the first patch is missing an Ack, @etienne-lms?

**Review by etienne-lms — COMMENTED** [review]
For commit "core: Add FFA_ID_GET handling for SPs":
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by jforissier** [comment]
@jellesels-arm please apply @etienne-lms's tag and I will merge this.

**Pull request merged**